dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/persister/AbstractTrackerPersister.java

```diff
@@ -320,8 +325,8 @@ private void assignFileResource( Session session, TrackerPreheat preheat, String
     protected void handleTrackedEntityAttributeValues( Session session, TrackerPreheat preheat,
         List<Attribute> payloadAttributes, TrackedEntityInstance trackedEntityInstance )
     {
-        Map<String, TrackedEntityAttributeValue> attributeValueDBMap = trackedEntityInstance
-            .getTrackedEntityAttributeValues()
+        Map<String, TrackedEntityAttributeValue> attributeValueDBMap = attributeValueService
+            .getTrackedEntityAttributeValues( trackedEntityInstance )
```

Comment:
Thinking out loud. Shouldnt the existing value teav fetches be part of preheating logic?

Reply:
You are right, there is something fishy going on here.
We are not preheating teav, we preheat TEI that has teav linked to it.
But then if we are persising a TEI and then an enrollment, we need those teav updated.
Using the service here, at this moment is the easiest solution but we will need to review the whole logic around teav.
There is an open discussion on how we should deal with TEI teav and Program teav, when this is over, we will need to come back to this code and possibly remove that service from there.

I will leave a comment in the code to remember it